<!DOCTYPE html>
<html lang="en-US">
<!--********************************************-->
<!--*       Generated from PreTeXt source      *-->
<!--*                                          *-->
<!--*         https://pretextbook.org          *-->
<!--*                                          *-->
<!--********************************************-->
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="robots" content="noindex, nofollow">
</head>
<body class="ignore-math">
<h3 class="heading"><span class="type">Paragraph</span></h3>
<div class="para logical"><ol class="decimal">
<li><div class="para">The first call is a failure because our specification demands that the result <code class="code-inline tex2jax_ignore">r1</code> must be 3, it is however 6.</div></li>
<li><div class="para">is not a failure.</div></li>
<li><div class="para">causes the program to get stuck. The last iteration of the loop in <code class="code-inline tex2jax_ignore">sum_first_n</code> intends to access the array element <code class="code-inline tex2jax_ignore">arr + 3</code> which does not exist and therefore the address <code class="code-inline tex2jax_ignore">arr + 3</code> is not valid. We discussed <a href="" class="xref" data-knowl="./knowl/bem-corr-stuck.html" title="Remark 7.2.2">before 7.2.2</a> that we cannot make any assumptions on the program's behavior in this case.</div></li>
</ol></div>
<span class="incontext"><a href="sec-corr-failures.html#p-758" class="internal">in-context</a></span>
</body>
</html>
